lib/matplotlib/tests/test_axes.py

```diff
@@ -4972,3 +4972,23 @@ def test_bar_single_height():
     ax.bar(range(4), 1)
     # Check that a horizontal chart with one width works
     ax.bar(0, 1, bottom=range(4), width=1, orientation='horizontal')
+
+
+def test_invalid_axes_limits():
```

Comment:
I think you should try to parametrize the tests too:

a first pass would be something as simple as:

```python
@pytest.mark.parametrize('value', [np.inf, np.nan])
def test_invalid_axes_limits(value)
    with pytest.raises(ValueError):
        plt.xlim(left=value)

    with pytest.raises(ValueError):
        plt.xlim(right=value)

    with pytest.raises(ValueError):
        plt.ylim(bottom=value)

    with pytest.raises(ValueError):
        plt.ylim(top=value)
```

Going two steps further, you could parametrize the limit functions and sides, but that might be going a bit too far:

```python
@pytest.mark.parametrize('value', [np.inf, np.nan])
@pytest.mark.parametrize(('key', 'setter'), [
    (plt.xlim, 'left'),
    (plt.xlim, 'right'),
    (plt.ylim, 'bottom'),
    (plt.ylim, 'top'),
])
def test_invalid_axes_limits(key, setter, value)
    limit = {key: value}
    with pytest.raises(ValueError):
        setter(**limit)
```